packages/cli-kit/src/public/node/ruby.ts

```diff
@@ -254,14 +256,10 @@ async function createThemeCheckCLIWorkingDirectory(): Promise<void> {
  * It creates the Gemfile to install The Ruby CLI and the dependencies.
  */
 async function createShopifyCLIGemfile(): Promise<void> {
-  const gemPath = joinPath(await shopifyCLIDirectory(), 'Gemfile')
-  const gemFileContent = ["source 'https://rubygems.org'", `gem 'shopify-cli', '${RubyCLIVersion}'`]
-  const {platform} = platformAndArch()
-  if (platform === 'windows') {
-    // 'wdm' is required by 'listen', see https://github.com/Shopify/cli/issues/780
-    gemFileContent.push("gem 'wdm', '>= 0.1.0'")
-  }
-  await file.writeFile(gemPath, gemFileContent.join('\n'))
+  const directory = await shopifyCLIDirectory()
+  await addContentToGemfile(directory, "source 'https://rubygems.org'")
+  await addContentToGemfile(directory, `gem 'shopify-cli', '${RubyCLIVersion}'`)
+  await addWindowsDependencies(directory)
```

Comment:
This can be one IO operation instead of 3:
- I create the content of the Gemfile in memory (as a string)
- Then I write the content to the `Gemfile` creating it if doesn't exist.